index.js

```diff
@@ -1,50 +1,78 @@
 'use strict';
 
-const wait = (store, actions, options) => {
-    options = Object.assign({
-        timeoutMs: 50,
-        actionArgument: undefined,
-    }, options);
-
-    if (!Array.isArray(actions)) {
-        actions = [actions];
+const isString = require('lodash/isString');
+const isSymbol = require('lodash/isSymbol');
+const isPlainObject = require('lodash/isPlainObject');
+const isEmpty = require('lodash/isEmpty');
+const differenceWith = require('lodash/differenceWith');
+
+function isValidAction(value) {
+    return isPlainObject(value) && (isString(value.type) || isSymbol(value.type));
+}
+
+function isValidActionType(value) {
+    return isString(value) || isSymbol(value);
+}
+
+const wait = (store, actions, timeout) => {
+    if (!timeout) {
+        timeout = 50;
     }
 
-    const originalDispatch = store.dispatch;
-    const unhookDispatch = () => {
-        store.dispatch = originalDispatch;
-    };
-    const complete = (timeout, callback) => {
-        clearTimeout(timeout);
-        unhookDispatch();
+    if (typeof actions !== 'function') {
```

Comment:
I think we can make a lot of the code below shorter, I will talk to you tomorrow.